### ISSUE TITLE
[RFC]: apply new migration when running yarn rw dev 

### ISSUE THREAD

**Issue by simoncrypta** [redwoodjs/redwood#7625]
### Summary

Check for a new migration when running `yarn rw dev` apply new migration or prob to reset database. It can be done with `yarn rw prisma migrate dev` 
Ref: https://www.prisma.io/docs/guides/database/developing-with-prisma-migrate/team-development

### Motivation

While working in a team, you may pull new changes in the database that you are not aware. Most of the time you sync with main, start your dev, get an error, find out that you need to run the new migration and waste 5 min of your time to realize this.

### Detailed proposal

Currently, we generate the Prisma client when we run `yarn rw dev`, we can change it for migration dev, so it would check for a new migration and generate the Prisma client. 

The output will be : 
```
Already in sync, no schema change or pending migration was found.

✔ Generated Prisma Client (4.9.0 | library
```
Or 
```
The following migration(s) have been applied:
[...]
Your database is now in sync with your schema.
✔ Generated Prisma Client (4.9.0 | library
```

I also don't feel that this will slow down the process 

### Are you interested in working on this?

- [X] I'm interested in working on this

**Comment by pantheredeye** [comment]
I like the motivation behind this. Taking any pain out of teamwork would be a good thing.

**Comment by dac09** [comment]
I think the only time this would be annoying is if you are _choosing_ not to run the migration. For example, switching between branches because you quickly wanted to check something. 

I think adding a prompt when migrations are available would be perfect, but not sure how we can detect umigrated migrations!

**Comment by simoncrypta** [comment]
Good point @dac09 I will try to play around to get a nice flow when review/switch branches.

**Comment by simoncrypta** [comment]
@dac09 I tried to do quick review with `yarn rw prisma migrate dev` and if you redo the same command back on main branch without the new migration, your database resync with the current state. I don't know when you're choosing to not run the migration and doing `yarn rw dev`, most of the time you will get errors. Also, I think is quick enough to not be a problem.

**Comment by simoncrypta** [comment]
Something we need to take in consideration is MongoDB don't support migration.

**Comment by Tobbe** [comment]
https://www.prisma.io/docs/reference/api-reference/command-reference#migrate-status and https://www.prisma.io/docs/reference/api-reference/command-reference#migrate-diff might come in handy here
